src/features/transfer-ownership/TransferOwnershipForm/TransferOwnershipForm.test.tsx

```diff
@@ -0,0 +1,465 @@
+// Setup //
+///////////
+const jestConsole = console;
+const consoleError = console.error;
+
+beforeEach(() => {
+	global.console = require('console');
+	console.error = jest.fn();
+});
+
+afterEach(() => {
+	global.console = jestConsole;
+	console.error = consoleError;
+	jest.clearAllMocks();
+});
+
+const renderComponent = (mockOwner?: string) =>
+	render(
+		<TransferOwnershipForm
+			domainId={mock.domain.id}
+			domainName={mock.domain.name}
+			domainTitle={mock.metadata.title}
+			domainOwner={mockOwner ? mockOwner : mock.domain.owner}
+			domainCreator={mock.domain.minter}
+			onClose={mockOnClose}
+		/>,
+	);
+
+const testSetup = () => {
+	return renderComponent();
+};
+
+/////////////
+// Helpers //
+/////////////
+const confirmScreenText =
+	'This transaction is about to be seared upon the blockchain. There’s no going back.';
+
+const onSubmitValidDetails = async () => {
+	const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+	await act(async () => {
+		fireEvent.input(input, { target: { value: mock.validInputAddress } });
+	});
+
+	fireEvent.click(
+		await screen.findByRole('button', {
+			name: /transfer/i,
+		}),
+	);
+};
+
+///////////
+// Tests //
+///////////
+describe('TransferOwnershipForm', () => {
+	// successful transfer ownership
+	test('should handle successful transfer domain ownership request', async () => {
+		mockTranferDomainOwnership.mockResolvedValue({ wait: mockTx });
+		mockTx.mockResolvedValue(undefined);
+		testSetup();
+
+		// details step
+		const onConfirmInputButton = await screen.findByRole('button', {
+			name: /transfer/i,
+		});
+
+		expect(onConfirmInputButton).toBeInTheDocument();
+		expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+		const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+		expect(input).toBeInTheDocument();
+
+		await act(async () => {
+			fireEvent.input(input, { target: { value: mock.validInputAddress } });
+		});
+
+		expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+		fireEvent.click(onConfirmInputButton);
+
+		// confirm step
+		const onConfirmTransactionButton = await screen.findByText(/confirm/i);
+
+		expect(onConfirmTransactionButton).toBeInTheDocument();
+		expect(onConfirmTransactionButton).not.toHaveAttribute('aria-disabled');
+
+		fireEvent.click(onConfirmTransactionButton);
+
+		// transaction approval step
+		screen.getByText('Please accept wallet transaction..');
+
+		// transaction in progress step
+		await screen.findByText('Your transaction is being processed...');
+
+		// assert successful request
+		await waitFor(() => {
+			expect(mockTranferDomainOwnership).toBeCalledTimes(1);
+			expect(mockTx).toBeCalledTimes(1);
+			expect(mockTranferDomainOwnership).toHaveBeenCalledWith(
+				mock.validInputAddress,
+				mock.domain.id,
+				{ isSigner: true },
+			);
+		});
+	});
+
+	describe('Details Step', () => {
+		test('should display correct domain data', async () => {
+			testSetup();
```

Comment:
Hm I left screen in for the helper rather than add props. Let me know if better to remove and pass the renderComponent selectors to the helper instead

Reply:
Disregard the above, `screen.getBy` (etc) is the correct pattern.